commons/src/main/java/org/infinispan/commons/marshall/MarshallableLambdaExternalizers.java

```diff
@@ -0,0 +1,235 @@
+
+   private static final int SET_VALUE_RETURN_PREV_OR_NULL = 1 | VALUE_MATCH_ALWAYS;
+   private static final int SET_VALUE_RETURN_VIEW = 2 | VALUE_MATCH_ALWAYS;
+   private static final int SET_VALUE_IF_ABSENT_RETURN_PREV_OR_NULL = 3 | VALUE_MATCH_EXPECTED;
+   private static final int SET_VALUE_IF_ABSENT_RETURN_BOOLEAN = 4 | VALUE_MATCH_EXPECTED;
+   private static final int SET_VALUE_IF_PRESENT_RETURN_PREV_OR_NULL = 5 | VALUE_MATCH_NON_NULL;
+   private static final int SET_VALUE_IF_PRESENT_RETURN_BOOLEAN = 6  | VALUE_MATCH_NON_NULL;
+   private static final int REMOVE_RETURN_PREV_OR_NULL = 7 | VALUE_MATCH_ALWAYS;
+   private static final int REMOVE_RETURN_BOOLEAN = 8 | VALUE_MATCH_ALWAYS;
+   private static final int REMOVE_IF_VALUE_EQUALS_RETURN_BOOLEAN = 9 | VALUE_MATCH_EXPECTED;
+   private static final int SET_VALUE_CONSUMER = 10 | VALUE_MATCH_ALWAYS;
+   private static final int REMOVE_CONSUMER = 11  | VALUE_MATCH_ALWAYS;
+   private static final int RETURN_READ_WRITE_FIND = 12 | VALUE_MATCH_ALWAYS;
+   private static final int RETURN_READ_WRITE_GET = 13 | VALUE_MATCH_ALWAYS;
+   private static final int RETURN_READ_WRITE_VIEW = 14 | VALUE_MATCH_ALWAYS;
+
+   public static final class ConstantLambdaExternalizer implements LambdaExternalizer<Object> {
+      private final IdentityIntMap<Class<?>> numbers = new IdentityIntMap<>(16);
+
+      public ConstantLambdaExternalizer() {
+         numbers.put(setValueReturnPrevOrNull().getClass(), SET_VALUE_RETURN_PREV_OR_NULL);
+         numbers.put(setValueReturnView().getClass(), SET_VALUE_RETURN_VIEW);
+         numbers.put(setValueIfAbsentReturnPrevOrNull().getClass(), SET_VALUE_IF_ABSENT_RETURN_PREV_OR_NULL);
+         numbers.put(setValueIfAbsentReturnBoolean().getClass(), SET_VALUE_IF_ABSENT_RETURN_BOOLEAN);
+         numbers.put(setValueIfPresentReturnPrevOrNull().getClass(), SET_VALUE_IF_PRESENT_RETURN_PREV_OR_NULL);
+         numbers.put(setValueIfPresentReturnBoolean().getClass(), SET_VALUE_IF_PRESENT_RETURN_BOOLEAN);
+         numbers.put(removeReturnPrevOrNull().getClass(), REMOVE_RETURN_PREV_OR_NULL);
+         numbers.put(removeReturnBoolean().getClass(), REMOVE_RETURN_BOOLEAN);
+         numbers.put(removeIfValueEqualsReturnBoolean().getClass(), REMOVE_IF_VALUE_EQUALS_RETURN_BOOLEAN);
+         numbers.put(setValueConsumer().getClass(), SET_VALUE_CONSUMER);
+         numbers.put(removeConsumer().getClass(), REMOVE_CONSUMER);
+         numbers.put(returnReadWriteFind().getClass(), RETURN_READ_WRITE_FIND);
+         numbers.put(returnReadWriteGet().getClass(), RETURN_READ_WRITE_GET);
+         numbers.put(returnReadWriteView().getClass(), RETURN_READ_WRITE_VIEW);
+      }
+
+      @Override
+      public ValueMatcherMode valueMatcher(Object o) {
+         int i = numbers.get(o.getClass(), -1);
+         if (i > 0) {
+            int valueMatcherId = ((i & VALUE_MATCH_MASK) >> 12) - 1;
+            return ValueMatcherMode.values()[valueMatcherId];
```

Comment:
This looks too fragile to me, the value matcher constants in this file can get out of sync with the `ValueMatcher` enum values.